src/optimizer/rule/simplification.rs

```diff
@@ -109,6 +117,91 @@ impl Rule for SimplifyFilter {
+                    if let ScalarExpression::Constant(value) = right_expr.as_ref() {
+                        match value.as_ref() {
+                            DataValue::Utf8(val_str) => {
+                                let mut value = val_str.clone().unwrap_or_else(|| "".to_string());
+
+                                if value.ends_with('%') {
+                                    value.pop(); // remove '%'
+                                    if let Some(last_char) = value.clone().pop() {
+                                        if let Some(next_char) = increment_char(last_char) {
+                                            let mut new_value = value.clone();
+                                            new_value.pop();
+                                            new_value.push(next_char);
+
+                                            let new_expr = ScalarExpression::Binary {
+                                                op: BinaryOperator::And,
+                                                left_expr: Box::new(ScalarExpression::Binary {
+                                                    op: BinaryOperator::GtEq,
+                                                    left_expr: left_expr.clone(),
+                                                    right_expr: Box::new(
+                                                        ScalarExpression::Constant(ValueRef::from(
+                                                            DataValue::Utf8(Some(value)),
+                                                        )),
+                                                    ),
+                                                    ty: ty.clone(),
+                                                }),
+                                                right_expr: Box::new(ScalarExpression::Binary {
+                                                    op: BinaryOperator::Lt,
+                                                    left_expr: left_expr.clone(),
+                                                    right_expr: Box::new(
+                                                        ScalarExpression::Constant(ValueRef::from(
+                                                            DataValue::Utf8(Some(new_value)),
+                                                        )),
+                                                    ),
+                                                    ty: ty.clone(),
+                                                }),
+                                                ty: ty.clone(),
+                                            };
+                                            filter_op.predicate = new_expr;
+                                        }
+                                    }
+                                }
+                            }
+                            _ => {
+                                graph.version += 1;
```

Comment:
Version + 1 is not required, this case only applies to global rules, e.g. ColumnPuring, line189 too